projects/kit/components/calendar-range/calendar-range.component.ts

```diff
@@ -104,8 +107,15 @@ export class TuiCalendarRangeComponent implements TuiWithOptionalMinMax<TuiDay>
             });
     }
 
-    readonly monthShiftMapper: TuiMapper<TuiMonth, TuiMonth> = item =>
-        item.append({month: 1});
+    @HostListener(`document:keydown.capture`, [`$event`])
+    onEsc(event: KeyboardEvent): void {
+        if (event.key !== `Escape` || !this.value?.isSingleDay) {
+            return;
+        }
+
+        event.stopPropagation();
+        this.value = this.previousValue;
```

Comment:
Looks like a bug =/

1. Select any date-range
2. Begin to select new range (click on any day once) and move mouse to any another day
3. Press Esc (cancel unfinished range selection)

**Actual behavour:**
Unfinished range selection is canceled but the previous finished one is selected.


https://user-images.githubusercontent.com/35179038/189603967-013cc6fb-6738-47ef-8dab-8a0709a91647.mov



**Expected behaviour:**
Unfinished range selection is canceled. No range is selected.

Reply:
need E2E test

Reply:
We have discussed this case with our designer.
It concludes that it is expected behaviour =)